### ISSUE TITLE
IOS: The application crashes if you try to download a file containing Cyrillic characters in the name

### ISSUE THREAD

**Issue by sergeykorablin** [zulip/zulip-mobile#2113]
In Android application everything is fine.

**Comment by borisyankov** [comment]
Hmm, interesting. This is an appropriate bug to fix for new contributors.

**Comment by gokcan** [comment]
Need to debug this. 
@zulipbot claim

**Comment by zulipbot** [comment]
Welcome to Zulip, @gokcan! We just sent you an invite to collaborate on this repository at https://github.com/zulip/zulip-mobile/invitations. Please accept this invite in order to claim this issue and begin a fun, rewarding experience contributing to Zulip!

Here's some tips to get you off to a good start:
* Join me on the [Zulip developers' server](https://chat.zulip.org), to get help, chat about this issue, and meet the other developers.
* Sign the [Dropbox Contributor License Agreement](https://opensource.dropbox.com/cla/), so that Zulip can use your code.
* [Unwatch this repository](https://help.github.com/articles/unwatching-repositories/), so that you don't get 100 emails a day.

As you work on this issue, you'll also want to refer to the [Zulip code contribution guide](https://zulip.readthedocs.io/en/latest/contributing/index.html), as well as the rest of the developer documentation on that site.

See you on the other side (that is, the pull request side)!

**Comment by gnprice** [comment]
The results of some debugging on this are in #2246 , but that PR was never finished.

I described a plan for how to fix this bug in this comment: https://github.com/zulip/zulip-mobile/pull/2246#issuecomment-380640640 with some additional details in the comment just after it.

**Comment by gnprice** [comment]
I couldn't reproduce this when I tried just now. This was in v26.28.151 (the current beta), on iOS 12.4.1 on an iPhone 7.

Specifically:
* I went to this message on czo: https://chat.zulip.org/#narrow/stream/7-test-here/topic/ContainsNonASCII/near/559041
  which has an uploaded image with Cyrillic as well as Latin letters in its filename.
* All was well. I tapped the image to open the lightbox.
* The image didn't actually show up! So that seems like there's still a related bug here.
* But the lightbox header and footer showed up just fine and remained responsive as normal.
* I tapped the overflow menu at the lower right, and "Download image".
* Got a toast saying the download succeeded.
* And indeed if I go over to the system Photos app, the image is there.

We haven't heard any new reports of this crash recently that I can recall, either.

Looking back at #2246 which was described as fixing this, it seems the details were actually all about getting the image to show up in the lightbox, too -- not about downloading, or any crash.
